core/src/test/java/org/apache/calcite/test/RelBuilderTest.java

```diff
@@ -3426,4 +3426,24 @@ private void checkExpandTable(RelBuilder builder, Matcher<RelNode> matcher) {
             builder.literal(5));
     assertThat(call.toStringRaw(), is("BETWEEN ASYMMETRIC($0, 1, 5)"));
   }
+
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-3926">[CALCITE-3926]
+   * CannotPlanException when an empty LogicalValues requires a certain collation</a>. */
+  @Test void testEmptyValuesWithCollation() throws Exception {
+    final RelBuilder builder = RelBuilder.create(config().build());
+    final RelNode root =
+        builder
+            .scan("DEPT")
+            .filter(builder.literal(false))
+            .sort(
+                builder.field("DNAME"),
+                builder.field("DEPTNO"))
+            .build();
+    try (PreparedStatement preparedStatement = RelRunners.run(root)) {
+      final String s = CalciteAssert.toString(preparedStatement.executeQuery());
+      final String result = "";
+      assertThat(s, is(result));
```

Comment:
Can you also compare the plan?

Reply:
Done

Reply:
I mean the final plan generated by planner, not by the RelBuilder.